compiler/llvm/clangUtil.cpp

```diff
@@ -3843,21 +3883,19 @@ static void linkLibDevice() {
     }
   }
 
-  // libdevice is a CUDA-specific thing
   if (getGpuCodegenType() == GpuCodegenType::GPU_CG_NVIDIA_CUDA) {
-    // load libdevice as a new module
-    llvm::SMDiagnostic err;
-    auto libdevice = llvm::parseIRFile(CHPL_CUDA_LIBDEVICE_PATH, err,
-                                       info->llvmContext);
-    //
-    // adjust it
-    const llvm::Triple &Triple = info->clangInfo->Clang->getTarget().getTriple();
-    libdevice->setTargetTriple(Triple.getTriple());
-    libdevice->setDataLayout(info->clangInfo->asmTargetLayoutStr);
-
-    // link
-    llvm::Linker::linkModules(*info->module, std::move(libdevice),
-                              llvm::Linker::Flags::LinkOnlyNeeded);
+    linkBitCodeFile(CHPL_CUDA_LIBDEVICE_PATH);
+  } else {
+    auto libPath = CHPL_ROCM_PATH + std::string("/amdgcn/bitcode");
+    linkBitCodeFile((libPath + "/hip.bc").c_str());
+    linkBitCodeFile((libPath + "/ocml.bc").c_str());
+    linkBitCodeFile((libPath + "/ockl.bc").c_str());
+    linkBitCodeFile((libPath + "/oclc_daz_opt_off.bc").c_str());
+    linkBitCodeFile((libPath + "/oclc_unsafe_math_off.bc").c_str());
+    linkBitCodeFile((libPath + "/oclc_finite_only_off.bc").c_str());
+    linkBitCodeFile((libPath + "/oclc_correctly_rounded_sqrt_on.bc").c_str());
```

Comment:
this is an amazing name for a library

Reply:
For reference there are "on" and "off" versions of several of these ".bc" files. I think the idea is they define some symbol which affects how other .bc files get linked in.

So basically you can parameterize what you want based off which of these you link to. I picked this particular set because that's what I see with `hipcc` if I pass `-###` and look at what they do.